### PR TITLE
[flah_ctrl/dv] Small 2 changes regarding flash_ctrl ral.set_hdl_path

### PR DESCRIPTION
**Pull request by eshapira**
Hi,

In this PR I added 2 small changes:
* Return **ral.set_hdl_path** in **flash_ctrl_env** to be independent of implementation, assuming extending envs will just use the same paths.
* Move flash_ctrl path in **ral.set_hdl_path** to come from **tb** through **uvm_config_db** rather then to be set specifically in **flash_ctrl_env**.

Thanks!

Signed-off-by: Eitan Shapira <eitan.shapira@nuvoton.com>

### CODE REVIEW

File: hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv [hunk 1]
```diff
@@ -16,6 +16,8 @@ class flash_ctrl_env #(type CFG_T = flash_ctrl_env_cfg,
   tl_agent        m_eflash_tl_agent;
   tl_reg_adapter  m_eflash_tl_reg_adapter;
 
+  string flash_ctrl_path;
```
Comment (sriyerg) [inline hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv:19]: nit: 
```suggestion
  string hdl_path_root;
```

File: hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv [hunk 2]
```diff
@@ -30,6 +32,10 @@ class flash_ctrl_env #(type CFG_T = flash_ctrl_env_cfg,
       cfg.m_eflash_tl_agent_cfg.d_ready_delay_min = 0;
       cfg.m_eflash_tl_agent_cfg.d_ready_delay_max = 0;
     end
+    // Retrieve flash_ctrl_path from tb for ral.
+    if (!uvm_config_db#(string)::get(this, "", "flash_ctrl_path", flash_ctrl_path)) begin
```
Comment (sriyerg) [inline hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv:36]:
```suggestion
    if (!uvm_config_db#(string)::get(this, "", "hdl_path_root", hdl_path_root)) begin
```

File: hw/ip/flash_ctrl/dv/tb/tb.sv [hunk 3]
```diff
@@ -183,6 +187,7 @@ module tb;
     uvm_config_db#(virtual tl_if)::set(null, "*.env.m_tl_agent_flash_ctrl_core_reg_block*", "vif",
                                        tl_if);
     uvm_config_db#(virtual tl_if)::set(null, "*.env.m_eflash_tl_agent*", "vif", eflash_tl_if);
+    uvm_config_db#(string)::set(null, "*.env", "flash_ctrl_path", `FLASH_CTRL_HIER_STR);
```
Comment (sriyerg) [inline hw/ip/flash_ctrl/dv/tb/tb.sv:190]: Could you move this into `gen_generic` block? Then there would be no need to define the macro `FLASH_CTRL_HIER_STR`. Directly set the string path as arg to `set()`.

File: hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv (hunk 1 above)
Comment (eshapira) [reply hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv]: Done.

File: hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv (hunk 2 above)
Comment (eshapira) [reply hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv]: Done.

File: hw/ip/flash_ctrl/dv/tb/tb.sv (hunk 3 above)
Comment (eshapira) [reply hw/ip/flash_ctrl/dv/tb/tb.sv]: Done.

Comment (eshapira) [reply hw/ip/flash_ctrl/dv/tb/tb.sv]: I can remove the macro, but this set can't be inside the gen_generic block, since this setting is needed also when I run it from the partner env (when the implementation isn't generic).

**Comment by eshapira** [comment]
Can you tell me the reason for the failing CI? It's declared private so I can't see it.

File: hw/ip/flash_ctrl/dv/tb/tb.sv (hunk 3 above)
Comment (sriyerg) [reply hw/ip/flash_ctrl/dv/tb/tb.sv]: With `partner_top` now being a bind module, the paths are same for open and closed. This invocation no longer needs to be in the `gen_generic` block (since the string path is going to be the same) AND you don't need to define the macro either.

File: hw/ip/flash_ctrl/dv/tb/tb.sv [hunk 4]
```diff
@@ -131,6 +131,7 @@ module tb;
                  "u_info_mem.gen_generic.u_impl_generic.mem"}, i, j)
 
   if (`PRIM_DEFAULT_IMPL == prim_pkg::ImplGeneric) begin : gen_generic
+
```
Comment (sriyerg) [inline hw/ip/flash_ctrl/dv/tb/tb.sv:134]: nit: Could you please revert this change?

File: hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv [hunk 5]
```diff
@@ -30,6 +32,10 @@ class flash_ctrl_env #(type CFG_T = flash_ctrl_env_cfg,
       cfg.m_eflash_tl_agent_cfg.d_ready_delay_min = 0;
       cfg.m_eflash_tl_agent_cfg.d_ready_delay_max = 0;
     end
+    // Retrieve hdl_path_root from tb for ral.
+    if (!uvm_config_db#(string)::get(this, "", "hdl_path_root", hdl_path_root)) begin
+          `uvm_fatal(`gfn, $sformatf("failed to get hdl_path_root from uvm_config_db"))
```
Comment (sriyerg) [inline hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv:37]: nit: spacing is off
```suggestion
      `uvm_fatal(`gfn, $sformatf("failed to get hdl_path_root from uvm_config_db"))
```

**Review by sriyerg — APPROVED** [review]
This PR is ready to go in one you address 2 more nit comments. Thanks - I like that we are able to get it working with much fewer changes in the open source.

File: hw/ip/flash_ctrl/dv/tb/tb.sv (hunk 3 above)
Comment (eshapira) [reply hw/ip/flash_ctrl/dv/tb/tb.sv]: Yes, I think you said what I meant. Just to be sure, that means you approve it as it is now?

File: hw/ip/flash_ctrl/dv/tb/tb.sv (hunk 4 above)
Comment (eshapira) [reply hw/ip/flash_ctrl/dv/tb/tb.sv:134]: Done.

File: hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv (hunk 5 above)
Comment (eshapira) [reply hw/ip/flash_ctrl/dv/env/flash_ctrl_env.sv:37]: Done.

**Comment by eshapira** [comment]
All done. Thanks!

**Review by sriyerg — APPROVED** [review]
